app/fight.py

```diff
@@ -0,0 +1,16 @@
+from typing import List
+
+from app.knight import Knight
+
+
+def fight(knights: List[Knight]):
+    for count, knight in enumerate(knights):
+        knight.hp += knight.protection
+
+        if count == 0:
+            knight.hp -= knights[count + 1].power
+        else:
+            knight.hp -= knights[count - 1].power

```

Comment:
This is so strange. And works wrong because if order of knights will be another it stoped works. You must fight by name